bears/general/AnnotationBear.py

```diff
@@ -0,0 +1,172 @@
+        and yields them as HiddenResults with the appropriate ranges.
+        Invalid comments or strings are:
+        1. Comments within comments,
+        2. Comments within strings and
+        3. Strings within comments.
+        """
+        file = ''.join(file)
+        annot_dict = {}
+        annot_dict['comment'] = {}
+        annot_dict['string'] = {}
+        annot_dict['string']['multiline'] = {}
+        annot_dict['string']['singleline'] = {}
+        annot_dict['comment']["singleline"] = {}
+        annot_dict['comment']["multiline"] = {}
+        # Tuple of tuples containing start and end of annotations
+        match_pos = ()
+        lang_dict = LanguageDefinition(language, language_family)
+
+        annot_dict['comment']["singleline"].update(
+                    lang_dict["comment_delimiter"])
+
+        annot_dict['comment']["multiline"].update(
+                    lang_dict["multiline_comment_delimiters"])
+
+        annot_dict['string']['singleline'].update(
+                                            lang_dict["string_delimiters"])
+        annot_dict['string']['multiline'].update(
+                                       lang_dict["multiline_string_delimiters"])
+        multi_str = single_str = multi_comm = ()
+        search_dict = annot_dict['string']['multiline']
+        multi_str = find_string_comment(file, search_dict, True)
+        search_dict = annot_dict['string']['singleline']
+        single_str = find_string_comment(file, search_dict, True)
+        search_dict = annot_dict['comment']["multiline"]
+        multi_comm = find_string_comment(file, search_dict, False)
+
+        match_pos += single_str + multi_str + multi_comm
+        # Seperate since single-line comments don't have a definite start and
+        # end
+        search_dict = annot_dict['comment']["singleline"]
+        single_comm = find_singleline_comments(file, search_dict, match_pos)
+        match_pos += single_comm
+        match_pos = remove_nested(match_pos)
+        m_comm = s_comm = m_str = s_str = []
+        if match_pos:
+            m_comm = list(get_range(file, filename, match_pos, multi_comm))
+            s_comm = list(get_range(file, filename, match_pos, single_comm))
+            m_str = list(get_range(file, filename, match_pos, multi_str))
+            s_str = list(get_range(file, filename, match_pos, single_str))
+        yield HiddenResult(self, m_comm)
+        yield HiddenResult(self, s_comm)
+        yield HiddenResult(self, m_str)
+        yield HiddenResult(self, s_str)
+
+
+def get_range(file, filename, match_pos, string_comment):
+    """
+    Checks if range is valid and then yields it.
+    """
+    search_range = []
+    for i in string_comment:
+        if i in match_pos:
+            search_range.append(i)
+    search_range = [(calc_line_col(file, start), calc_line_col(file, end))
+                    for (start, end) in search_range]
+
+    for i in search_range:
```

Comment:
imo it's really better to provide a meaningful name, one-letter-names are reserved for indices (at least for me :P)